source-code/core/src/lint/rules/additionalKey.ts

```diff
@@ -0,0 +1,24 @@
+import type { Context } from '../context.js'
+import { createRule } from '../rule.js'
+
+export const additionalKeyRule = createRule('inlang.additionalKey', () => {
+	let context: Context
+	let referenceLanguage: string
+
+	return {
+		initialize: (config) => {
+			context = config.context
+			referenceLanguage = config.referenceLanguage
+		},
```

Comment:
I wonder if we can abstract more with the `createRule` utility (which seems like a great idea!). 

Initialize seems boilerplaty. For example, the `createRule` utility could pass the config and env (?) as arguments to the function. 

```ts
export const additionalKeyRule = createRule('inlang.additionalKey', (config) => {

	return {
        // 
    }
```

Reply:
No it can't. You have to think what will be called by the regular `JavaScript` code (when you import the `inlang.config.js` file) and what get's called by the `lint` function. When configuring a rule, we can't inject any `inlang` concepts.
Unless we would do it like this:
```diff
config = {
-	lint: [
+ 	lint: () => [
 		additionalKeyRule('warning')
 	]
}
```
But I think this will cause more irritation as it is unusual for config files.

Reply:
> what get's called by the lint function.  

The lint function could inject the config. But as stated in the summary, I think this API "is good enough".

Reply:
> The lint function could inject the config.

That's exactly what happens in the `initialize` function